packages/medusa/src/interfaces/payment-service.ts

```diff
@@ -0,0 +1,89 @@
+import { Payment } from "@models/payment"
+import { Cart } from "@models/cart"
+import { PaymentSession, PaymentSessionStatus } from "@models/payment-session"
+import { Customer } from "@models/customer"
+import { BaseService } from "medusa-interfaces"
+
+export type Data = Record<string, unknown>
+export type PaymentData = Data
+export type PaymentSessionData = Data
+
+export interface PaymentService {
+  getIdentifier(): string
+
+  onCreatePaymentSession(cart: Cart): Promise<PaymentSessionData>
+
+  onCreatePayment(paymentSession: PaymentSession): Promise<PaymentData>
+
+  retrievePayment(cart: Cart): Promise<Payment>
+
+  updatePayment(
+    paymentSessionData: PaymentSessionData,
+    cart: Cart
+  ): Promise<PaymentSessionData>
+
+  authorizePayment(
+    paymentSession: PaymentSession,
+    context: Data
+  ): Promise<{ data: PaymentSessionData; status: PaymentSessionStatus }>
+
+  capturePayment(payment: Payment): Promise<PaymentData>
+
+  refundPayment(payment: Payment, refundAmount: number): Promise<PaymentData>
+
+  deletePayment(paymentSession: PaymentSession): Promise<void>
```

Comment:
`onDeletePayment`

Reply:
this one is called with `refreshSession` so probably keeping `deletePayment` to keep the action as it is ?